pkg/validation/options.go

```diff
@@ -287,6 +287,9 @@ func parseProviderInfo(o *options.Options, msgs []string) []string {
 	case *providers.GitLabProvider:
 		p.AllowUnverifiedEmail = o.InsecureOIDCAllowUnverifiedEmail
 		p.Groups = o.GitLabGroup
+		if len(o.GitlabProjects) > 0 {
```

Comment:
NIT: I think this logic belongs in `SetProjects` around the scope manipulation.  Just to keep things together, it is detached from the area driving the reason for this logic when over here.

Avoids the double standard since we are perfectly fine setting the parallel functionality `p.Groups` to an empty array right above this.

Reply:
If I clearly understand what you say, here is the fix : https://github.com/oauth2-proxy/oauth2-proxy/pull/630/commits/542056e57c72408526cab1dc5de63e2d3919a087

Reply:
Unfortunately, I think that commit brings us back to the original problem (scope is manipulated even when we don't need to).

This is what I was thinking:
`if len(o.GitlabProjects) > 0 && !strings.Contains(p.Scope, "read_api")`

Just to keep the logic together and only manipulate `scope` if needed.